di2f/index.md

```diff
@@ -0,0 +1,226 @@
+
+- Buckets
+  - Mutable filesystem API
+  - Pinning API
+- A `go-datastore` interface (key-value store)
+- Web-hooks
+- Filecoin
+  - Thread anchoring
+  - Bucket archiving
+  - Deal retrievals
+- Databases
+  - MongoDB (e.g., direct connection URI)
+  - PostgreSQL
+- Media encoding
+- etc.
+
+The diagram in [@fig:filecoin-service] shows how a pinning service (via Buckets) and Filecoin can be orchestrated as peer services.
+
+![Buckets and Filecoin can be considered network services.](./assets/filecoin-service.png){#fig:filecoin-service}
+
+In addition to exposing more "traditional" web2 APIs (e.g., REST, gRPC, etc.) as services, it is possible to expose these APIs over p2p protocols. For example, it is possible to allow peers to serve HTTP endpoints and make HTTP requests through `libp2p` using Go's standard "http" and "net" stack. This provides a simple on-ramp for web 2.0 developers to expose their APIs over the threads p2p network. Couple service provision and remote database access with the robust authentication and globally identifiable assets afforded by thread DIDs, and we now have a very clear path to onboarding web2 developers (and data) to the decentralized web.
+
+# Conclusions
+
+Decentralized identifiers (DIDs) enable cross-protocol/blockchain interactions by defining a common interface to retrieve and validate entities, such as users and data. A DID-driven threads network offers an exciting opportunity to expose p2p services as first-class components of a p2p system.
+
+Threads and IPFS/Filecoin benefit from a native DID in numerous ways. Firstly, DIDs provide globally unique namespaces for resources, such that thread IDs act as identifiers to network-wide resources. Secondly, adopting a standardized system for resource identification increases ecosystem(s) interoperability, facilitating integration with external DID methods available on systems such as Filecoin or Ethereum, as well as a range of identity solutions (e.g., Ceramic/IDX). Thirdly, globally unique identifiers pave the way for _distributed authorization_ to network resources, because access control mechanisms are globally and unambiguously defined. This leads to a net increase in the decentralization of network services.
+
+An additional key benefit of the proposed system includes the provision of "native" discovery mechanisms. Network peers can do work on each others' behalf since any peer is capable of discovering and resolving thread DID namespaces. Additionally, since all operations on a thread lead to deterministic, content-addressable updates, any peer on the network can participate in the update of a thread to which they have access.
+
+<!-- @sanderpick this is pretty hand-wavy here, needs some help! -->
```

Comment:
@sanderpick this is pretty hand-wavy here, needs some help!

Reply:
I think you're hitting the points well! Perhaps we can mention something about service discovery, and how that enables a kind of market. I'm thinking here about a "market of broker", one level up from the "deal market" we've been chatting about around NEAR.